### DISCUSSION TITLE
@RestClient configuration

### DISCUSSION THREAD

**Discussion by ennishol** [quarkusio/quarkus#36423] [Q&A]
How does RestClient work internally? Is there a connection pool? If so, can it be configured and how?

Are there any limitations for the number of connections? These things can be configured where using Apache http client library, but what about RestClient classic and also reactive?

Thanks!

**Comment by mapuci** [comment]
all configurations are available here: https://quarkus.io/guides/all-config
search for `rest-client`, you will find means to configure basically everything you can configure with apache itself.

If you are using `quarkus-rest-client-reactive` you are using `vertex-web-client`. 
If you are using `quarkus-rest-client`you are using `org.apache.httpcomponents:httpclient`
No matter what you use, both libs are wrapped with microprofile rest client spec, so both are configured with same properties.

If you like programatic approach, see https://quarkus.io/guides/rest-client-reactive#programmatic-client-creation-with-quarkusrestclientbuilder
